cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +139,17 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
-
+  private Properties generateKafkaConfig(CConfiguration cConf) {
     Properties prop = new Properties();
-    prop.setProperty("broker.id", Integer.toString(brokerId));
-    if (hostname != null) {
-      prop.setProperty("host.name", hostname);
+
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..server\\.*)$");
+    for (Map.Entry<String, String> pair : propConfigs.entrySet()) {
+      String key = pair.getKey();
+      String trimmedKey = key.substring(13);
+      prop.setProperty(trimmedKey, pair.getValue());
     }
-    prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("socket.send.buffer.bytes", "1048576");
-    prop.setProperty("socket.receive.buffer.bytes", "1048576");
-    prop.setProperty("socket.request.max.bytes", "104857600");
-    prop.setProperty("log.dir", logDir);
-    prop.setProperty("num.partitions", Integer.toString(numPartitions));
-    prop.setProperty("log.retention.hours", "24");
-    prop.setProperty("log.flush.interval.messages", "10000");
-    prop.setProperty("log.flush.interval.ms", "1000");
-    prop.setProperty("log.segment.bytes", "536870912");
-    prop.setProperty("zookeeper.connect", zkConnectStr);
-    prop.setProperty("zookeeper.connection.timeout.ms", "1000000");
-    prop.setProperty("default.replication.factor", Integer.toString(replicationFactor));
+    Preconditions.checkState(Integer.parseInt(prop.getProperty(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG)) > 0,
```

Comment:
You either move the port check here or move this check out. All checking should be in one place.